### PR TITLE
Issue 1577: UNKNOWN domain expired bug

### LINKED ISSUE #1577: Domain is showing "Expired" when it is in Unknown state
### Current Behavior

All domains will start in the **Unknown** state when not made in the registry yet. 

When a domain manager just had their application approved, their domain is in **Unknown** status and thus has no expiration date. 

Currently, when the user goes to the homepage, they will see their brand new domain as “**Expired**”. This is misleading.

### Expected Behavior

A domain shows “**Expired**”, only if it is **NOT** in the **Unknown** state.

### Steps to Reproduce

1. Login as Domain manager 
2. make sure you have a domain for which you haven't visited Manage Domains yet (thus it will be in an unknown state)
3. View domains
4. see the one unknown domain is actually showing as expired

### Environment

_No response_

### Additional Context
note: if you don't have any unknown domains anymore (due to past testing) do the following
1. manually make a domain in django admin and add yourself as creator
2. in Django admin User Domain Roles add a new role for yourself on this particular domain
3. no go back to view domains and see that it is expire

Alternatively, you can approve one of your submitted applications which will more accurately simulate how a user will see a domain appear after an approval

### Issue Links

_No response_

### PR DESCRIPTION
**Pull request by rachidatecs**
## Ticket

Resolves #1577 

## Changes

- Do not show expired for UNKNOWN domains on homepage
- Do not show expired for UNKNOWN domains on domain detail page
- Unit tests

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/templates/domain_detail.html [hunk 1]
```diff
@@ -17,9 +17,10 @@
           <span class="text-bold text-primary-darker">
             Status:
           </span>
-            {% if domain.is_expired %}
+            {# UNKNOWN domains would not have an expiration date and thus would show 'Expired' #}
+            {% if domain.is_expired and domain.state != domain.State.UNKNOWN %}
               Expired
-            {% elif domain.state == domain.State.UNKNOWN or domain.state == domain.State.DNS_NEEDED%}
+            {% elif domain.state == domain.State.UNKNOWN or domain.state == domain.State.DNS_NEEDED %}
```
Comment (neilmb) [inline src/registrar/templates/domain_detail.html:23]: **question (non-blocking)**: A designer can probably help us, but is "DNS needed" the right thing to display here if a domain is in our system but it has never been visited with EPP so it shows up as "UNKNOWN"?

**Review by neilmb — APPROVED** [review]
Just one honest question, otherwise great work.

File: src/registrar/templates/domain_detail.html (hunk 1 above)
Comment (rachidatecs) [reply src/registrar/templates/domain_detail.html:23]: Good question. I want to say this has been debated already but can't remember which ticket had an AC to display DNS Needed for Unknown domains. Pretty sure it came from an AC.

**Review by therealslimhsiehdy — APPROVED** [review]

File: src/registrar/templates/domain_detail.html (hunk 1 above)
Comment (abroddrick) [reply src/registrar/templates/domain_detail.html:23]: That was a request from them actually. They didn't want it to show as Unknown or Unset or whatever. They said user won't care, show the same label basically

**Pull request merged**